artiq/language/environment.py

```diff
@@ -209,18 +209,28 @@ def get(self, key, processor, group, tooltip):
         self.requested_args[key] = processor, group, tooltip
         return None
 
+    def check_unprocessed_arguments(self):
+        pass
 
 class ProcessArgumentManager:
     def __init__(self, unprocessed_arguments):
         self.unprocessed_arguments = unprocessed_arguments
+        self._processed_arguments = []
 
     def get(self, key, processor, group, tooltip):
         if key in self.unprocessed_arguments:
             r = processor.process(self.unprocessed_arguments[key])
+            self._processed_arguments.append(key)
         else:
             r = processor.default()
         return r
 
+    def check_unprocessed_arguments(self):
+        unprocessed = set(self.unprocessed_arguments.keys()) -\
+                      set(self._processed_arguments)
+        if unprocessed:
+            raise AttributeError("Unprocessed argument(s): " +
```

Comment:
This is not a very user-friendly message, since the "argument processor" is not documented or exposed. Something like "Invalid arguments: ..." would be better